File: modules/itac/jvm/src/main/java-igmore/edu/gemini/tac/qengine/qengine/skycalc/DecBinCalc.java
/*
 * Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
 * For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 */

//
// $
//

package edu.gemini.qengine.skycalc;

import edu.gemini.spModel.core.Semester;
import edu.gemini.spModel.core.Site;
import edu.gemini.skycalc.Angle;

import java.util.Date;
import java.util.List;

/**
 * Describes the contract for calculating the DecBin percentages.
 */
public interface DecBinCalc {

    List<Percent> calc(Site site, Date start, Date end, DecBinSize size, Angle ra);
}

File: modules/itac/src/main/scala/skycalc/DecBinCalc.scala
// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause

package edu.gemini.qengine.skycalc;

import edu.gemini.qengine.skycalc.DecBinSize
import edu.gemini.tac.qengine.util.Percent
import lucuma.core.enums.Site
import lucuma.core.math.RightAscension

import java.time.Instant

trait DecBinCalc:
  def calc(site: Site, start: Instant, end: Instant, size: DecBinSize, ra: RightAscension): List[Percent] 